enterprise/backend/test/metabase_enterprise/harbormaster/client_test.clj

```diff
@@ -0,0 +1,28 @@
+(ns metabase-enterprise.harbormaster.client-test
+  (:require
+   [clojure.test :refer [deftest is testing]]
+   [metabase-enterprise.harbormaster.client :as hm.client]
+   [metabase.models.setting :as setting]
+   [metabase.test.util :as mt]))
+
+(deftest ->config-good-test
+  (testing "Both needed values are present and pulled from settings"
+    (mt/with-temporary-setting-values
+      [api-key "mb_api_key_123"
+       store-api-url "http://store-api-url.com"]
+      (is (= {:store-api-url "http://store-api-url.com", :api-key "mb_api_key_123"}
+             (#'hm.client/->config))))))
+
+(deftest ->config-blank-api-key-test
+  (let [grv setting/get-raw-value]
+    ;; mt/with-temporary-setting-values is not used here because we want to test the behavior of the function when the
+    ;; api-key is blank or nil, and mt/with-temporary-setting-values will not allow us to set the api-key to blank
```

Comment:
would `mt/with-temporary-raw-setting-values` work here?

Reply:
works for "", fails for nil. 🫠  I'll replace the "" case